File: Shells/Cpp/sudoku/Odometer.cpp
#include"Odometer.hpp"

int odometerToInt(std::string value) {
    int retVal = 0;
    const std::string* found = std::find(std::begin(ODOMETER), std::end(ODOMETER), value);
    if(found != ODOMETER + 36){
        retVal = found - ODOMETER;
    } else {
        std::cout<<"odometerToInt failed. Invlaid value "<<value<<"Replaced with 0"<<std::endl;
        retVal = 0;
    }
    return retVal;
}

std::string intToOdometer(int num){
    std::string retVal;
    try{
        retVal = ODOMETER[num];
    }catch(const std::out_of_range& oor){
        std::cout<<"intToOdometer failed. Invalid number "<<num <<" Replaced with 0"<<std::endl;
        retVal = "0";
    }
    return retVal;
}

File: Sudoku_Java_Shell/src/sudoku/Odometer.java
package sudoku;
import java.util.Arrays;


public class Odometer {
	public static int odometerToInt(String value) {
		int retVal = Arrays.binarySearch(ODOMETER, value.toUpperCase());
		if(retVal < 0)
		{
			System.err.println("odometerToInt failed. Invlaid value " + value + "Replaced with 0");
			retVal = 0;
		}
		return retVal;
	}
	
	public static String intToOdometer(int num)
	{
		String retVal;
		try{
			retVal = ODOMETER[num];
		}catch(IndexOutOfBoundsException e){
			System.err.println("intToOdometer failed. Invalid number " + num + " Replaced with 0");
			retVal = "0";
		}
		return retVal;
	}

	private static final String[] ODOMETER = 
		{"0","1","2","3","4","5","6","7","8","9",
		 "A","B","C","D","E","F","G","H","I","J",
		 "K","L","M","N","O","P","Q","R","S","T",
		 "U","V","W","X","Y","Z"};
}
